plugins/uefi/fu-uefi-udisks.c

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright (C) 2019 Mario Limonciello <mario.limonciello@dell.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <gio/gio.h>
+
+#include "fu-uefi-udisks.h"
+#include "fwupd-common.h"
+#include "fwupd-error.h"
+
+#define UDISKS_DBUS_SERVICE		"org.freedesktop.UDisks2"
+#define UDISKS_DBUS_PATH		"/org/freedesktop/UDisks2/Manager"
+#define UDISKS_DBUS_MANAGER_INTERFACE	"org.freedesktop.UDisks2.Manager"
+#define UDISKS_DBUS_PART_INTERFACE 	"org.freedesktop.UDisks2.Partition"
+#define UDISKS_DBUS_FILE_INTERFACE	"org.freedesktop.UDisks2.Filesystem"
+#define ESP_DISK_TYPE			"c12a7328-f81f-11d2-ba4b-00a0c93ec93b"
+
+gboolean
+fu_uefi_is_udisks_path (const gchar *path)
+{
+	return g_str_has_prefix (path, "/org/freedesktop/UDisks2/");
+}
+
+static GDBusProxy *
+fu_uefi_get_dbus_proxy (const gchar *path, const gchar *interface,
+			GError **error)
+{
+	g_autoptr(GDBusConnection) connection = NULL;
+	g_autoptr(GDBusProxy) proxy = NULL;
+
+	connection = g_bus_get_sync (G_BUS_TYPE_SYSTEM, NULL, error);
+	if (connection == NULL) {
+		g_prefix_error (error, "failed to get bus: ");
+		return NULL;
+	}
+	proxy = g_dbus_proxy_new_sync (connection,
+				       G_DBUS_PROXY_FLAGS_NONE, NULL,
+				       UDISKS_DBUS_SERVICE,
+				       path,
+				       interface,
+				       NULL, error);
+	if (proxy == NULL) {
+		g_prefix_error (error, "failed to find %s: ", UDISKS_DBUS_SERVICE);
+		return NULL;
+	}
+	return g_steal_pointer (&proxy);
+}
+
+GVariant *
+fu_uefi_get_block_devices (GError **error)
```

Comment:
Not sure I approve of "exporting" a GVariant -- can't we convert that to something friendlier? e.g. GPtrArray of gchar -- a GVariant is somewhat of an implementation detail, and it's tricky to express how to sink, or unref it.

Reply:
fixed